linkpreview/preview/base.py

```diff
@@ -7,9 +9,20 @@ class PreviewBase(object):  # pragma: nocover
     Base for all web preview.
     """
 
-    def __init__(self, link: Link, parser: str):
+    def __init__(self, link: Link, parser: Union[str, None], soup: Union[BeautifulSoup, None]):
+        if parser and soup:
+            raise Exception(
+                'Only one of `parser` or `soup` argument must be provided to PreviewBase')
+
+        if not parser and not soup:
```

Comment:
@biolds This kind of validation is not necessary. bs4 is fine without passing `parser`, and also the developer will receive proper exception if not specifing the `soup`.